azure-iot-device/azure/iot/device/provisioning/abstract_provisioning_device_client.py

```diff
@@ -0,0 +1,79 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+This module provides an abstract interface representing clients which can communicate with the
+Device Provisioning Service.
+"""
+
+import abc
+import six
+import logging
+from .security.sk_security_client import SymmetricKeySecurityClient
+from azure.iot.device.provisioning.pipeline.provisioning_pipeline import ProvisioningPipeline
+
+logger = logging.getLogger(__name__)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class AbstractProvisioningDeviceClient(object):
+    """
+    Super class for any client that can be used to register devices to Device Provisioning Service.
+    """
+
+    def __init__(self, provisioning_pipeline):
+        """
+        Initializes the provisioning client.
+        :param provisioning_pipeline: Instance of the provisioning pipeline object.
+        """
+        self._provisioning_pipeline = provisioning_pipeline
+
+    @classmethod
+    def create_from_security_client(cls, security_client, protocol_name):
```

Comment:
> def create_from_security_client(cls, security_client, protocol_name): [](start = 3, length = 70)

For now I think this works, but I suspect we'll have to revisit the client architecture entirely when X509 comes along.

Reply:
yes that is the next step

---
In reply to: [289559936](https://github.com/Azure/azure-iot-sdk-python-preview/pull/98#discussion_r289559936) [](ancestors = 289559936)